solutions/bronze/usaco-1324.mdx

```diff
@@ -0,0 +1,302 @@
+---
+id: usaco-1324
+source: USACO Bronze 2023 US Open
+title: Moo Language
+author: Kevin Sheng
+---
+
+[Official Editorial (C++, Python)](https://usaco.org/current/data/sol_prob2_bronze_open23.html)
+
+## Explanation
+
+The official editorial covers a linear time approach, but quadratic is sufficient for full credit.
+
+This problem certainly is a doozy, though.
+The way to think about it is to pin down what you want to set as constant,
+then see how the others can be defined in terms of the things that you've already set.
+
+In this case, the constants we'll brute force over are the number of transitive and intransitive verbs,
+which I'll call TVs and ITVs for brevity.
+The intuitive reason for this is that each sentence, not counting conjuctions, has exactly one verb.
+Thus, if we know how many verbs we're going to use, we can know how many nouns, commas, etc. we're going to use as well.
+
+TVs and ITVs need two and one noun respectively, so we have to check if we have enough nouns first.
+If we use a nonzero number of TVs, we can use commas to append nouns at the end of a TV sentence.
+
+We also try to use as many conjuctions as possible;
+if there's $s$ sentences, we can use a maximum of $\left\lfloor\frac{s}{2}\right\rfloor$ conjuctions,
+since each sentence can be joined at most one time.
+
+The rest is some pretty annoying formatting, but it doesn't require any algorithmic tricks.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <vector>
+
+using namespace std;
+
+int main() {
+	int test_num;
+	cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		int word_num, comma_num, period_num;
+		cin >> word_num >> comma_num >> period_num;
+		map<string, vector<string>> words{{"noun", {}},
+		                                  {"transitive-verb", {}},
+		                                  {"intransitive-verb", {}},
+		                                  {"conjunction", {}}};
+		for (int w = 0; w < word_num; w++) {
+			string token, type;
+			cin >> token >> type;
+			words.at(type).push_back(token);
+		}
+
+		int max_size = 0;
+		int best_tv = 0;
+		int best_itv = 0;
+		const vector<string> &nouns = words["noun"];  // just a shorthand
+		int max_verbs = period_num + min(period_num, (int)words["conjunction"].size());
+		// Test all combinations of transntive/intransitive verbs
+		for (int tv = 0; tv <= words["transitive-verb"].size(); tv++) {
+			for (int itv = 0; itv <= words["intransitive-verb"].size(); itv++) {
+				int nouns_left = nouns.size() - itv - 2 * tv;
+				if (nouns_left < 0 || tv + itv > max_verbs) { break; }
+				int conj_used = min((int)words["conjunction"].size(), (tv + itv) / 2);
+				int extra = tv > 0 ? min(comma_num, nouns_left) : 0;
```

Comment:
overall commenting is a bit weak i think. it's nice to explain where these math formulas come from